convert2rhel/unit_tests/actions/pre_ponr_changes/kernel_modules_test.py

```diff
@@ -181,6 +181,12 @@ def test_ensure_compatibility_of_kmods_check_env(
         " We will continue the conversion with the following kernel modules unavailable in RHEL:.*"
     )
     assert re.match(pattern=should_be_in_logs, string=caplog.records[-1].message, flags=re.MULTILINE | re.DOTALL)
+    # cannot assert exact action message contents as the kmods arrangement in the message is not static
+    assert "level=WARNING" in str(ensure_kernel_modules_compatibility_instance.messages)
+    assert "id=ALLOW_UNAVAILABLE_KERNEL_MODULES" in str(ensure_kernel_modules_compatibility_instance.messages)
+    assert "Detected 'CONVERT2RHEL_ALLOW_UNAVAILABLE_KMODS' environment variable." in str(
+        ensure_kernel_modules_compatibility_instance.messages
+    )
```

Comment:
You could do it also like this: 

```python
    message = ensure_kernel_modules_compatibility_instance.messages[0]
    assert actions.STATUS_CODE["WARNING"] == message.level
    assert "ALLOW_UNAVAILABLE_KERNEL_MODULES" == message.id
    assert "Detected 'CONVERT2RHEL_ALLOW_UNAVAILABLE_KMODS' environment variable." in message.message
```

Which, in my opinion, is way more readable as well

Reply:
Since in this case you have a list of results, I would say that you don't necessarily need to on `assert_actions_result` for that assertation, and rather, you can do it manually like the above suggestion.

Reply:
Or you could also try to use the same strategy of your other checks using `set`

```python
    expected = set(
        (
            actions.ActionMessage(
                level="INFO",
                id="PACKAGE_UPDATES_CHECK_SKIP_NO_PUBLIC_REPOSITORIES",
                message=(
                    "Skipping the check because there are no publicly available Oracle Linux Server 8.6 repositories available."
                ),
            ),
        )
    )

    assert expected.issuperset(package_updates_action.messages)
    assert expected.issubset(package_updates_action.messages)
```

Reply:
I would do it with the subsets but the message inside of the ActionMessage is not static so it won't pass most of the time